lib/segment/src/index/query_optimization/payload_provider.rs

```diff
@@ -52,7 +52,7 @@ impl PayloadProvider {
             // Rewrite condition checking code to support error reporting.
             // Which may lead to slowdown and assumes a lot of changes.
             PayloadStorageEnum::OnDiskPayloadStorage(s) => s
-                .read_payload(point_id)
+                .read_payload(point_id, &hw_counter)
```

Comment:
`hw_counter` should be initialized in the proper place

Reply:
What do you mean? There is a TODO for propagating the measured value which will be done in a separate PR.
If you want me to implement it here, I can do that too. I'm just trying to keep PRs small

Reply:
the PR is called

> Finish io measurement for payload storage

Either we need to implement it, or rename the PR

Reply:
Renamed it 👍🏻